x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/update_signals.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { defaults } from 'lodash/fp';
+import { AlertAction } from '../../../../../alerting/server/types';
+import { AlertsClient } from '../../../../../alerting/server/alerts_client';
+import { ActionsClient } from '../../../../../actions/server/actions_client';
+import { readSignals } from './read_signals';
+
+export interface SignalParams {
+  alertsClient: AlertsClient;
+  actionsClient: ActionsClient;
+  description?: string;
+  from?: string;
+  id: string;
+  index?: string[];
+  interval?: string;
+  enabled?: boolean;
+  filter?: Record<string, {}> | undefined;
+  kql?: string | undefined;
+  maxSignals?: string;
+  name?: string;
+  severity?: number;
+  type?: string; // TODO: Replace this type with a static enum type
+  to?: string;
+  references?: string[];
+}
+
+export const calculateInterval = (
```

Comment:
This is a good candidate for some of our first unit tests since it is a pure function.

I will write one for this.